src/core/vertexmodel.cpp

```diff
@@ -696,6 +727,25 @@ void VertexModel::setCurrentPoint( const QgsPoint &point )
 
   Vertex &vertex = mVertices[mCurrentIndex];
 
+  // Calculate angles between edges
+
+  int startPoint = mCurrentIndex - 2;
+  if ( startPoint < 0 )
+  {
+    startPoint = vertexCount() + startPoint;
+  }
+  int endPoint = mCurrentIndex + 2;
+  if ( endPoint >= vertexCount() )
+  {
+    endPoint = endPoint - vertexCount();
+  }
+
+  double angle = calculateAngle( mVertices[startPoint].point, mVertices[mCurrentIndex].point, mVertices[endPoint].point );
```

Comment:
Alright, as discussed yesterday, here I think we shouldn't calculate the angle but rather the point location of the point that'd form the angle we're looking for. 

Then, we will snap to that point whenever we are close to it.